src/services/repo/useRepoSettings.tsx

```diff
@@ -0,0 +1,151 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+import A from 'ui/A'
+
+import { RepoNotFoundErrorSchema } from './schemas/RepoNotFoundError'
+import { RepoOwnerNotActivatedErrorSchema } from './schemas/RepoOwnerNotActivatedError'
+
+
+const RepositorySchema = z.object({
+  __typename: z.literal('Repository'),
+  private: z.boolean().nullable(),
+  activated: z.boolean().nullable(),
+  uploadToken: z.string().nullable(),
+  defaultBranch: z.string().nullable(),
+  profilingToken: z.string().nullable(),
+  staticAnalysisToken: z.string().nullable(),
+  graphToken: z.string().nullable(),
+  yaml: z.string().nullable(),
+  bot: z
+    .object({
+      username: z.string().nullable(),
+    })
+    .nullable(),
+})
+
+interface FetchRepoSettingsArgs {
+  provider: string
+  owner: string
+  repo: string
+  signal?: AbortSignal
+}
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      repository: z
+        .discriminatedUnion('__typename', [
+          RepositorySchema,
+          RepoNotFoundErrorSchema,
+          RepoOwnerNotActivatedErrorSchema,
+        ])
+        .nullable(),
+    })
+    .nullable(),
+})
+
+function fetchRepoSettingsDetails({
+  provider,
+  owner,
+  repo,
+  signal,
+}: FetchRepoSettingsArgs) {
+  const query = `
+    query GetRepoSettings($name: String!, $repo: String!) {
+      owner(username:$name){
+        repository(name:$repo) {
+          __typename
+          ... on Repository {
+            private
+            activated
+            uploadToken
+            defaultBranch
+            profilingToken
+            staticAnalysisToken
+            graphToken
+            yaml
+            bot {
+              username
+            }
+          }
+          ... on NotFoundError {
+            message
+          }
+          ... on OwnerNotActivatedError {
+            message
+          }
+        }
+      }
+    }
+`
+  return Api.graphql({
+    provider,
+    query,
+    signal,
+    variables: {
+      name: owner,
+      repo,
+    },
+  }).then((res) => {
+    const parsedRes = RequestSchema.safeParse(res?.data)
+
+    if (!parsedRes.success) {
+      return Promise.reject({
+        status: 404,
+        data: {},
+        dev: 'useRepoSettings - 404 schema parsing failed',
```

Comment:
Could we similarly add this "dev" key to the NotFound and OwnerNotActivated error cases?